_source/_posts/2022-09-15-kafka-microservices.md

```diff
@@ -0,0 +1,757 @@
+...
+```
+
+Add the inbound binding configuration to `application.yml`:
+
+```yaml
+spring:
+  cloud:
+    stream:
+      bindings:
+        ...
+        binding-in-store-alert:
+          destination: store-alerts-topic
+          content-type: application/json
+          group: store-alerts
+```
+
+Create an `EmailService` to send the store update notification, using the Spring Framework's `JavaMailSender`.
+
+```java
+package com.okta.developer.alert.service;
+
+import com.okta.developer.alert.service.dto.StoreAlertDTO;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Service;
+
+@Service
+public class EmailService {
+
+    private JavaMailSender emailSender;
+
+    @Value("${alert.distribution-list}")
+    private String distributionList;
+
+    public EmailService(JavaMailSender emailSender){
+        this.emailSender = emailSender;
+    }
+
+    public void sendSimpleMessage(StoreAlertDTO alertDTO){
+        try {
+            SimpleMailMessage message = new SimpleMailMessage();
+            message.setTo(distributionList);
+            message.setSubject("Store Alert: " + alertDTO.getStoreName());
+            message.setText(alertDTO.getStoreStatus());
+            message.setFrom("StoreAlert");
+            emailSender.send(message);
+        } catch (Exception exception) {
+            throw new EmailServiceException(exception);
+        }
+    }
+}
+```
+
+Create the referenced `EmailServiceException`.
+
+```java
+package com.okta.developer.alert.service;
+
+public class EmailServiceException extends RuntimeException {
+
+    public EmailServiceException(Exception exception) {
+        super(exception);
+    }
+}
+```
+
+Add a `StoreAlertDTO` class in the `...service.dto` package.
+
+```java
+package com.okta.developer.alert.service.dto;
+
+public class StoreAlertDTO {
+
+    private String storeName;
+    private String storeStatus;
+
+    public String getStoreName() {
+        return storeName;
+    }
+
+    public void setStoreName(String storeName) {
+        this.storeName = storeName;
+    }
+
+    public String getStoreStatus() {
+        return storeStatus;
+    }
+
+    public void setStoreStatus(String storeStatus) {
+        this.storeStatus = storeStatus;
+    }
+
+}
+```
+
+Add a new property to `alert/src/main/resources/config/application.yml` **and** to `alert/src/test/resources/config/application.yml` for the destination email of the store alert.
+
+```yaml
+alert:
+  distribution-list: {distributionListAddress}
+```
+
+**NOTE:** You'll need to set a value for the email (e.g., `list@email.com` will work) in `src/test/.../application.yml` for tests to pass. For Docker, you'll override the `{distributionListAddress}` and `{username}` + `{password}` placeholder values with environment variables below.
```

Comment:
@havi You suggested `e.g.` instead of `e.g.,`. However, I've seen Grammarly suggest this will the command so I'm not sure which is correct.